File: lib/repeated_string.rb
class RepeatedString
  def initialize(string = '', string_size = 0)
    @string = string
    @string_size = string_size
  end

  def count
    matches = @string.scan('a').size
    times = @string_size / @string.size
    diff = @string_size - ( @string.size * times )
    if diff === 0
      matches * times
    else
      @str = ''
      @i = 0
      diff.times do
        @str += @string.split('')[@i]
        @i += 1
      end
      matches * times + @str.scan('a').size
    end
  end
end

File: src/repeatedString.js

const repeatedString = (str = '', size) => {
  const exp = new RegExp(/a/g);
  const matches = str.match(exp).length;
  const times = parseInt(size / str.length);
  const diff = size - ( str.length * times );

  if (diff > 0) {
      let rest = '';
      for(let i = 0; i < diff; i++) {
        rest += str.split('')[i];
      }
      let more_matches = rest.match(exp) || [];
      return ( matches * times ) + more_matches.length;
  }

  return matches * times;
};

module.exports = repeatedString;
